### DISCUSSION TITLE
ILI9488 with ESP32 - Rotation Issue

### DISCUSSION THREAD

**Discussion by Biocas2007** [Bodmer/TFT_eSPI#1892] [Q&A - General]
I am hoping one of the more talented members may be able to shed some light on the situation I am having with this display. For some reason I cannot seem to get the rotation to display correctly on the screen. The only rotation that works is tft.setRotation(1) which is always at the top left corner at (0,0). Any other rotation seems to take the top left corner and uses that point as the start. The following issues are noticed when changing the rotation.

Rotation(0) - portrait, all text is mirrored on top left point
Rotation(1) - landscape, All text fine
Rotation(2) - portrait, some text upside down, some text mirrored, some both
Rotation(3) - landscape, Text1 upside down and mirrored, TextFont7 fine, TextFont4 upside down and mirrored

The sketch used is TFT_Print_Test. Based on what seems to be happening is that the (0,0) coordinate does not seem to be changing to the correct location when the rotation is changed. If anyone has any insight on where I should begin my troubleshooting I would greatly appreciate it. Attached are the images relation to the issue with the rotation along with the model of the TFT screen.

![ILI9488](https://user-images.githubusercontent.com/59484534/176574086-04d83b64-985d-40ce-a51e-ef8882e99e9d.jpg)
![Rotation0](https://user-images.githubusercontent.com/59484534/176574088-4df8d330-ad6a-4db2-b814-fca03c00c177.jpg)
![Rotation1](https://user-images.githubusercontent.com/59484534/176574089-f5fe2cb8-b8f9-4857-8c58-4aceeb1a8785.jpg)
![Rotation2](https://user-images.githubusercontent.com/59484534/176574090-11c7dfa3-c946-48f9-9432-5fed31f54db7.jpg)
![Rotation3](https://user-images.githubusercontent.com/59484534/176574091-64918750-46f2-44eb-b5f8-bd529e5b27b1.jpg)

**Comment by Bodmer** [comment]
This type of display issue typically occurs if the wrong controller type is used. Provide a link to the vendors page.

**Reply by Biocas2007** [reply]
I purchased this from Amazon. This is the device. https://www.amazon.com/dp/B08CN6M9VC?psc=1&ref=ppx_yo2ov_dt_b_product_details. It states it is from Fafaicy model: ‎Fafeicya1846psrzg. I have done some research on this but cannot find any type of white paper about the device other than the defaults for how it is suppose to work. Nothing from Fafaicy though. I have also tried using other ILIxxxx drivers but they do not seem to work either. Everything else seems to work on it without an issue. I have been able to solder on the additional 4 ins and run the SD card, touch and TFT screen all at once with the ILI9488 drivers. It seems to just be with the rotation.
